TACC-Cloud/hazmapper
# task/WG-83: Handle Overlapping Point Clouds

## Overview: ##
Putting a point marker with the polygon/point cloud. This would make it be able to be seen if zoomed out and would allow users to select different overlapping point clouds. Adding those point markers to the marker cluster, as well.
## PR Status: ##


## Related Jira tickets: ##

* [WG-83](https://tacc-main.atlassian.net/browse/WG-83)

## Summary of Changes: ##

- Added a custom marker to the top-left corner of point clouds, so that users can click the icon and get the correct modal to pop up in cases where there were overlapping point clouds in the same location.
- Added the ability to attach feature data to the marker options prop (so that the modals open when the markers are clicked, which is consistent/same behavior as clicking inside the polygon area)
- Added function to calculate the top-left corner of a layer/feature (this could be refactored or added onto later for other corners)
- Added point cloud markers to the marker cluster group, so that all assets would be included in clusters.
- Added conditionals to the featureClickHandler function to check whether clicked object was a marker or layer.

## Testing Steps: ##
1. Go to localhost:4200 and find a project where there are overlapping point clouds present. (If you need a project, Nathan or I can share one with you)
2. Go to the overlapping point clouds on the map. Check that the new point-cloud icon is present in the top-left corner of the polygons.
3. Click the point-cloud icons and make sure the correct modals open for that point-cloud asset.
4. Zoom out and make sure you see cluster markers and corresponding numbers related to how many assets are in that area--including the overlapping point clouds.
5. Compare with images below

## UI Photos:
**New Point Cloud Markers**
<img width="1260" alt="Point_cloud_icons" src="https://github.com/TACC-Cloud/hazmapper/assets/50084480/936ad8b0-d78e-459f-ac39-f32b4ab6dd4b">
<br>
**Cluster markers with point-cloud assets included**
![cluster_marker2](https://github.com/TACC-Cloud/hazmapper/assets/50084480/efe8bdc4-4e23-400d-b1b3-d8e50db512e0)
![cluster_marker1](https://github.com/TACC-Cloud/hazmapper/assets/50084480/97a43226-a668-4fa7-b3c2-6c23458daa2e)


## Notes: ##
